.editorconfig

```diff
@@ -0,0 +1,20 @@
+# See https://editorconfig.org/
+
+# This is the top-most .editorconfig file
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.{cpp,h,mm,nsi,pro,ps1,py,sh}]
+indent_style = space
+indent_size = 4
+
+[*.{yaml,yml}]
+indent_style = space
+indent_size = 2
+
+[libs/**]
```

Comment:
There are no other parts which should be excluded? Are you sure (docs/)

Reply:
Hm, I don't see any obvious mistakes when editing a .md file from docs/ with vim (and the editorconfig plugin) with this config. I think the line ending/trailing LF/stripping trailing whitespace should apply as well (and works in my tests).

We should keep in mind that this is not about re-formatting existing files, but about keeping the style when creating new files or adding existing files (well, the counter example is `shfmt` which does use this file to decide about the proper formatting for `.sh` files, see #2582).

Reply:
Correct that this one is important when creating new files, but just as important, maybe even more important, is the gitconfig file.
See my latest comment on [this issue](https://github.com/jamulussoftware/jamulus/issues/2552#issuecomment-1090465840).
Without the proper gitconfig, on Windows all text files are converted to CR/LF on checkout.

Reply:
I think you should check the config for ps1 files. they are typical windows and might expect cr/lf (especially the older Powershells).

Reply:
> Correct that this one is important when creating new files, but just as important, maybe even more important, is the gitconfig file.

That's right, but [.git/config cannot be part of a repository](https://stackoverflow.com/questions/18329621/how-to-store-a-git-config-as-part-of-the-repository/18330114#18330114) AFAIK, so while this is related to this PR, it's not something we can do as part of the repo.

> I think you should check the config for ps1 files. they are typical windows and might expect cr/lf (especially the older Powershells).

They are already supposed to be LF, but might be converted to CRLF on Windows checkouts.
```
$ file .github/autobuild/windows.ps1 
.github/autobuild/windows.ps1: ASCII text
```

```
$ hexdump -c .github/autobuild/windows.ps1  | head -n4
0000000   #       S   t   e   p   s       f   o   r       g   e   n   e
0000010   r   a   t   i   n   g       W   i   n   d   o   w   s       a
0000020   r   t   i   f   a   c   t   s       v   i   a       G   i   t
0000030   h   u   b       A   c   t   i   o   n   s  \n   #       S   e

$ git show master:.github/autobuild/windows.ps1 | hexdump -c | head -n4
0000000   #       S   t   e   p   s       f   o   r       g   e   n   e
0000010   r   a   t   i   n   g       W   i   n   d   o   w   s       a
0000020   r   t   i   f   a   c   t   s       v   i   a       G   i   t
0000030   h   u   b       A   c   t   i   o   n   s  \n   #       S   e

$ git ls-files .github/autobuild/windows.ps1 windows/* --eol
i/lf    w/lf    attr/                 	.github/autobuild/windows.ps1
i/lf    w/lf    attr/                 	windows/NSISCopyRegistryKey/README.txt
i/lf    w/lf    attr/                 	windows/NSISCopyRegistryKey/registry.nsh
i/lf    w/lf    attr/                 	windows/deploy_windows.ps1
i/-text w/-text attr/                 	windows/installer-banner.bmp
i/-text w/-text attr/                 	windows/installer-welcome.bmp
i/lf    w/lf    attr/                 	windows/installer.nsi
i/-text w/-text attr/                 	windows/jamulus-server-icon-2020.ico
i/-text w/-text attr/                 	windows/mainicon.ico
i/lf    w/lf    attr/                 	windows/mainicon.rc
i/lf    w/lf    attr/                 	windows/nsProcess/COPYING
i/lf    w/lf    attr/                 	windows/nsProcess/ConvFunc.h
i/lf    w/lf    attr/                 	windows/nsProcess/README_JamulusChanges.txt
i/lf    w/lf    attr/                 	windows/nsProcess/Readme.txt
i/lf    w/lf    attr/                 	windows/nsProcess/api.h
i/lf    w/lf    attr/                 	windows/nsProcess/nsProcess.c
i/lf    w/lf    attr/                 	windows/nsProcess/nsProcess.sln
i/lf    w/lf    attr/                 	windows/nsProcess/nsProcess.vcxproj
i/lf    w/lf    attr/                 	windows/nsProcess/nsProcess.vcxproj.filters
i/lf    w/lf    attr/                 	windows/nsProcess/nsis_tchar.h
i/lf    w/lf    attr/                 	windows/nsProcess/pluginapi.c
i/lf    w/lf    attr/                 	windows/nsProcess/pluginapi.h
i/lf    w/lf    attr/                 	windows/sound.cpp
i/lf    w/lf    attr/                 	windows/sound.h
```

Reply:
> That's right, but [.git/config cannot be part of a repository](https://stackoverflow.com/questions/18329621/how-to-store-a-git-config-as-part-of-the-repository/18330114#18330114) AFAIK

That's right. But you **can** include a .gitconfig file in the root of the repo and let contributers execute `git config --local include.path ../.gitconfig` to include that one in .git/config.

Reply:
> That's right. But you **can** include a .gitconfig file in the root of the repo and let contributers execute `git config --local include.path ../.gitconfig` to include that one in .git/config.

Yes, sure, but it defeats the reasoning for Git not supporting that by default. :)

> EDIT: Apparently you also can use .gitattributes to do the same: See [here](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings) or [here](https://stackoverflow.com/questions/9976986/force-lf-eol-in-git-repo-and-working-copy).

Ah, I wasn't aware that those could be part of the repo. This would definitely make sense.